### ISSUE TITLE
Text alignment is incorrect

### ISSUE THREAD

**Issue by sharahamo** [kupiqu/plot2svg#13]
I am importing the figure from MATLAB and the text alignment is incorrect. This is how it looks like in InkScape: 
<img width="616" alt="screen shot 2018-11-03 at 12 25 58" src="https://user-images.githubusercontent.com/8395313/47952229-f5006400-df63-11e8-96b4-41abda00bcf2.png">

And in the file viewer, it is indeed a bit higher than it is supposed to be, for example: 

<img width="298" alt="screen shot 2018-11-03 at 12 28 56" src="https://user-images.githubusercontent.com/8395313/47952241-106b6f00-df64-11e8-8e7f-5b46e9d170ff.png">

I have to adjust the text to bring it lower when importing to pdf_tex. But what causes the problem?

**Comment by kupiqu** [comment]
Really no idea with just a screenshot (which is useful to see what the result is but not the cause), or in other words, if I knew this would have been probably fixed already.

So, please tell me your matlab version and give the most minimalistic code snippet you can get that reproduces the issue in your system for me to try reproducing and debugging the issue in mine.

**Comment by sharahamo** [comment]
Yes sorry was actually doing that but hit posted before actually finishing it. It can be as simple as this: 

`x=[1:10]
y=[1:10]
plot(x, y)
xlabel('some text that is not aligned')
xticks([1 5 10])
xticklabels({'some'; 'text'; 'arent aligned'})
plot2svg 'Figures/test.svg'`

It will be exaggerated if you set the font size: 

`set(gca, 'FontSize', 13,'LineWidth', 2, 'FontName', 'Roboto');`

I am using 2018b but I had this problem since 2016b

On MATLAB: 
<img width="552" alt="screen shot 2018-11-03 at 12 46 32" src="https://user-images.githubusercontent.com/8395313/47952397-91c40100-df66-11e8-90a8-cb9f4101adf7.png">

On viewer:
<img width="314" alt="screen shot 2018-11-03 at 12 46 44" src="https://user-images.githubusercontent.com/8395313/47952399-98eb0f00-df66-11e8-8317-7d8f001e8f47.png">

In inkscape: 
<img width="448" alt="screen shot 2018-11-03 at 12 48 31" src="https://user-images.githubusercontent.com/8395313/47952416-cdf76180-df66-11e8-8be9-fc4e5f07f970.png">

**Comment by kupiqu** [comment]
Thanks. I'll take a look... (text alignment is very tricky though, but it may be room for improvement).

**Comment by kupiqu** [comment]
Sorry, I cannot reproduce. It may be the problem is in your viewer. What viewer is it?
I tested this exact code:

```
x=[1:10];
y=[1:10];
plot(x, y)
xlabel('some text that is not aligned')
xticks([1 5 10])
xticklabels({'some'; 'text'; 'aren''t aligned'})
set(gca, 'FontSize', 13,'LineWidth', 2, 'FontName', 'Roboto')
plot2svg 'test.svg';
```
Matlab 2018b:

![image](https://user-images.githubusercontent.com/3875196/47952761-c3c76a00-df4a-11e8-97fc-6b038e847926.png)

Chromium:

![image](https://user-images.githubusercontent.com/3875196/47952732-6cc19500-df4a-11e8-83f9-933240886286.png)

Inkscape:

![image](https://user-images.githubusercontent.com/3875196/47952744-9b3f7000-df4a-11e8-98c3-2e7461a52752.png)

**Comment by sharahamo** [comment]
Does the alignment in inkscape works fine for you too? Because I have problem in inkscape, i dont really care about the viewer. I mean if you select the texts, is it selecting the text, or selecting the space above text? and which inkscape version are you using?

**Comment by kupiqu** [comment]
It looks fine in the screenshot, isn't it?

>

**Comment by sharahamo** [comment]
You'll have to highlight the text in inkscape for me to know. For example, here's using plot2svg, the node (small square) is above text and the highlighted part are above the text:

<img width="464" alt="screen shot 2018-11-03 at 14 12 57" src="https://user-images.githubusercontent.com/8395313/47953226-b45c1700-df72-11e8-9d9e-768ab39def62.png">

here's using MATLAB save as svg:

<img width="339" alt="screen shot 2018-11-03 at 14 13 22" src="https://user-images.githubusercontent.com/8395313/47953231-c2aa3300-df72-11e8-91c5-7cfc2e4ddbb9.png">

**Comment by kupiqu** [comment]
I see. That might be as it uses baseline shifts for alignment. Will take a
look when I get back to my computer...

**Comment by sharahamo** [comment]
My MATLAB's default vertical alignment is 'Top' for xlabel and any text I put. And for some reason, plot2svg is not converting the position correctly

**Comment by sharahamo** [comment]
Another problem is: when I hit enter on the text (in inkscape) to make it double like, the text is divided into two boxes? 

These problems do not exist when I tried using the old plot2svg. however, i do need to use the scatter function. much appreciated for your help!

**Comment by kupiqu** [comment]
> I see. That might be as it uses baseline shifts for alignment. Will take a look when I get back to my computer...

I can confirm this shift in the baseline. Unfortunately, this comes from how text is currently allocated in plot2svg. It's certainly not optimal, but didn't find a better way. Other options I tested created a visual offset, which is far from trivial to compensate for (different font sizes, axes locations, 2d vs 3d plots etc.).

So I will leave this issue open as a low priority bug for now. Feel free to submit a pull request if you happen to find a better solution.

**Comment by sharahamo** [comment]
I wonder if there's any explanation why the previous version of plot2svg works and if there's a simple workaround for this? it seems to happen in only xtick and xticklabel

**Comment by kupiqu** [comment]
> I wonder if there's any explanation why the previous version of plot2svg works and if there's a simple workaround for this? it seems to happen in only xtick and xticklabel

The previous version manually compensated for the offset, but I found it to be suboptimal under certain circumstances. This new version seemed to work better in general, although I wasn't aware of this current issue...

**Comment by kupiqu** [comment]
fixed with commit d8a4444ac6760940ba3355ed7e160d1fe52f91b4

**Comment by kupiqu** [comment]
Current implementation seems much better, @sharahamo please give it a try.
